hollow/src/main/java/com/netflix/hollow/api/producer/fs/HollowFilesystemPublisher.java

```diff
@@ -44,7 +44,30 @@ public HollowFilesystemPublisher(Path blobStorePath) {
     }
 
     @Override
-    public void publish(HollowProducer.Blob blob) {
+    public void publish(HollowProducer.AbstractPublishArtifact publishArtifact) {
+        if (publishArtifact instanceof HollowProducer.HeaderBlob) {
+            publishHeader((HollowProducer.HeaderBlob) publishArtifact);
+        } else {
+            publishBlob((HollowProducer.Blob) publishArtifact);
+        }
+    }
+
+    private void publishHeader(HollowProducer.HeaderBlob headerBlob) {
+        Path destination = blobStorePath.resolve(String.format("header-%d", headerBlob.getVersion()));
+        try (
+                InputStream is = headerBlob.newInputStream();
+                OutputStream os = Files.newOutputStream(destination)
+        ) {
+            byte buf[] = new byte[4096];
```

Comment:
Minor: the buffered copy code is now at 3 places in this file, consider moving it into a different method